dapps/profile.vue/src/components/did/DID.ts

```diff
@@ -143,14 +143,21 @@ export default class DIDComponent extends mixins(EvanComponent) {
    * @param endpoint updated endpoint
    */
   onUpdateEndpoint(index: number, endpoint: ServiceEndpoint): void {
+    // Replace with updated endpoint
     this.endpoints = this.endpoints.map((item, idx) => (idx === index ? endpoint : item));
-    if (!endpoint.id || !endpoint.url || !endpoint.type) {
+
+    if (!endpoint.id || !endpoint.url || !endpoint.type || !this.isEndpointsUnique) {
       this.canSave = false;
     } else {
       this.canSave = true;
     }
   }
 
+  get isEndpointsUnique(): boolean {
+    const ids = this.endpoints.map((endpoint) => endpoint.id);
+    return new Set(ids).size === this.endpoints.length;
```

Comment:
wouldn't it be more convenient to check each endpoint if there was already declared the same before and directly show at the duplicated endpoint an error?

Reply:
Yes, that's why it's in draft. My current WIP PoC is a modular approach to form validation.